File: java/LongestCommonPrefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder result = new StringBuilder();
        int i = 0;
        Boolean shortest = false;
        if (strs.length == 0) {
            return result.toString();
        }
        
        while (true == true) {
            for (int j = 0; j < strs.length; j ++) {
                if (i >= strs[j].length()) {
                    shortest = true;
                    break;
                }
            }
            if (shortest == true) {
                return result.toString();
            }
            char firstChar = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j ++) {
                if (strs[j].charAt(i) != firstChar) {
                    return result.toString();
                }
            }
            result.append(firstChar);
            i += 1;
        }
    }
}

File: archive/python/string/LongestCommonPrefix.py
class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        result = ""
        if len(strs) == 0:
            return result
            
        min_length = 10 ** 10
        for i in xrange(len(strs)):
            min_length = min(min_length, len(strs[i]))
            
        for i in xrange(min_length):
            curr = strs[0][i]
            for j in xrange(1, len(strs)):
                if strs[j][i] != curr:
                    return result
            result += curr
        
        return result
                
